lib/libutils/ext/trace.c

```diff
@@ -82,7 +82,7 @@ static int print_core_id(char *buf, size_t bs)
 	if (thread_get_exceptions() & THREAD_EXCP_FOREIGN_INTR)
 		return snprintk(buf, bs, "%0*zu ", num_digits, get_core_pos());
 	else
-		return snprintk(buf, bs, "%*s ", num_digits, "?");
+		return snprintk(buf, bs, "%*s ", 1, "?");
```

Comment:
I am not sure which alignment you are referring too. And yes other way could be to fix symbolize.py.

Reply:
If a CPU core number is printed instead of '?' it may take two digits. So the lines with '?' would not be aligned with the lines with a number.
That being said, I know realize that such lines can never be mixed inside the same message block, so that is actually not really a problem.

You know what? In the end I prefer your solution...

Would you please simplify the snprintk() to:
```C
		return snprintk(buf, bs, "?");
```

Reply:
Ok will simplify it.

Reply:
Given that we're fixing symbolize.py. What problem are we fixing here?

Reply:
The real fix would be this. Question mark always comes right after the column ':?' for all platforms. It just looks better IMO.
The symbolize.py change would therefore not be needed, although I think it is still useful for robustness as I said (but I can drop it if you prefer).

Reply:
How about two (or more) question marks to match with the number of digits needed to display the core number?

Reply:
I would like it equally or even a bit better :)
In this case `symbolize.py` should be modified in the same commit for consistency. And I would drop my other PR.
@b49020 are you OK with this approach?

Reply:
yeah it looks even better. Will implement it.